source/adapters/level_zero/command_buffer.cpp

```diff
@@ -759,21 +764,14 @@ UR_APIEXPORT ur_result_t UR_APICALL urCommandBufferEnqueueExp(
 
     UR_CALL(createEventAndAssociateQueue(Queue, &RetEvent,
                                          UR_COMMAND_COMMAND_BUFFER_ENQUEUE_EXP,
-                                         SignalCommandList, false));
+                                         SignalCommandList, false, true));
 
     ZE2UR_CALL(zeCommandListAppendBarrier,
                (SignalCommandList->first, RetEvent->ZeEvent, 1,
                 &(CommandBuffer->SignalEvent->ZeEvent)));
+    Queue->executeCommandList(SignalCommandList, false, false);
```

Comment:
Think we can move `ur_command_list_ptr_t SignalCommandList{};` declaration to inside the `if (Event)` scope now